backstory/src/main/java/com/google/sps/images/VisionManager.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.images;
+
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.protobuf.ByteString;
+import com.google.sps.images.data.AnnotatedImage;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ *
+ * VisionManager is an ImagesManager implemented using the Google Vision API.
+ * VisionManager Manages the gathering and packaging of Vision API image analytics.
+ */
+public final class VisionManager implements ImagesManager {
+  private final ImageAnnotatorClient imageAnnotatorClient;
+
+  /**
+   * Creates a vision manager object using the image annotator client from Vision API.
+   */
+  public VisionManager() throws IOException {
+    ImageAnnotatorClient imageAnnotatorClient = ImageAnnotatorClient.create();
+    this.imageAnnotatorClient = imageAnnotatorClient;
+  }
+
+  /**
+   * Creates a vision manager object using a mock image annotator client.
+   */
+  public VisionManager(ImageAnnotatorClient imageAnnotatorClient) {
+    this.imageAnnotatorClient = imageAnnotatorClient;
+  }
+
+  @Override
+  public List<AnnotatedImage> createAnnotatedImagesFromImagesAsByteArrays(
+      List<byte[]> imagesAsByteArrays) throws IOException {
+    List<AnnotatedImage> annotatedImages = new ArrayList<>();
+
+    for (byte[] rawImageData : imagesAsByteArrays) {
```

Comment:
Maybe add a TODO here to parallelize (its rare to make request serially unless they have an order dependency).

```
// TODO: parallelize calls to detectLabelsFromImageBytes() since it requires a network call.
```

Reply:
Absolutely right. I was thinking about this, but for the MVP (since we only allow one image upload anyways) we can keep the current implementation. Updated to add the to-do.